labs-ios-starter/Surveys/View/SingleRowPickerView.swift

```diff
@@ -0,0 +1,58 @@
+//
+//  SingleRowPickerView.swift
+//  labs-ios-starter
+//
+//  Created by Kenny on 10/1/20.
+//  Copyright © 2020 Lambda, Inc. All rights reserved.
+//
+
+import UIKit
+
+typealias CompletionHandler = () -> Void
+
+protocol SingleRowSpinnerDelegate: UIGestureRecognizerDelegate {
+    func updateSpinner()
+}
+
+class SingleRowPickerView: UIPickerView {
+
+    var rowHeight: CGFloat {
+        self.rowSize(forComponent: 0).height
+    }
+
+    var tapDelegate: SingleRowSpinnerDelegate? {
+        didSet {
+            commoninit()
```

Comment:
only configures tap gesture recognizer when delegate is assigned. I assume that's intentional

Reply:
Yep. I thought it was a pretty clever way of finding a place where:

a. both are guaranteed not nil
b. it doesn't delay operability (it's available to use as soon as you can see it)